features/steps/amun_api.py

```diff
@@ -0,0 +1,132 @@
+
+from amun import get_inspection_status
+from amun import get_inspection_build_log
+
+from behave import given, when, then
+
+
+@given("amun service is accessible using {scheme}")
+def amun_accessible(context, scheme):
+    """Check amun service is accessible using HTTP or HTTPS."""
+    if scheme not in ("HTTPS", "HTTP"):
+        raise ValueError(f"Invalid scheme {scheme!r}, has to be HTTP or HTTPS")
+
+    scheme = scheme.lower()
+    context.amun_api_host = f"{scheme}://{os.environ['THOTH_AMUN_API_HOST']}/api/v1"
+    response = requests.get(context.amun_api_host, verify=False)
+
+    assert (
+        response.status_code == 200
+    ), f"Invalid response when accessing Amun API /api/v1 endpoint: {response.status_code!r}"
+
+    assert response.text, "Empty response from server for Amun API /api/v1 endpoint"
+
+
+@when("an amun inspection job is scheduled")
+def amun_schedule(context):
+    """Schedule a test amun inspection job."""
+    configuration = Configuration()
+    configuration.host = context.amun_api_host
+    api_client = ApiClient(configuration)
+    api_instance = InspectionApi(api_client)
+    specification = InspectionSpecification(
+        allowed_failures=1,
+        base="registry.access.redhat.com/ubi8/python-36",
+        build=InspectionSpecificationBuild(
+            requests=InspectionSpecificationBuildRequests(
+                cpu="1",
+                hardware=InspectionSpecificationBuildRequestsHardware(
+                    cpu_family=6,
+                    cpu_model=94,
+                    physical_cpus=32,
+                    processor="Intel-Xeon-Processor-Skylake-IBRS"
+                ),
+                memory="1Gi"
+            )
+        ),
+        identifier="test",
+        packages=[
+            "vim"
+        ],
+        python_packages=[
+            "pipenv"
+        ],
+        run=InspectionSpecificationRun(
+            requests=InspectionSpecificationRunRequests(
+                cpu="1",
+                hardware=InspectionSpecificationRunRequestsHardware(
+                    cpu_family=6,
+                    cpu_model=94,
+                    physical_cpus=32,
+                    processor="Intel-Xeon-Processor-Skylake-IBRS"
+                ),
+                memory="512Mi"
+            )
+        ),
+        script="#!/usr/bin/bash\necho \"Here should be run tests...\"\n",
+        update=True
+    )
+    api_response = api_instance.post_inspection(specification)
+    context.inspection_id = api_response.to_dict()["inspection_id"]
+
+
+@then("wait for inspection to finsih successfully")
+def wait_for_inspection_to_finish(context):
+    """Wait for scheduled inspection to finish."""
+    retries = 0
+    while True:
+        if retries > timedelta(minutes=45).total_seconds():
+            raise RuntimeError("Inspection job took too much time to finish")
+
+        response = get_inspection_status(context.amun_api_host, context.inspection_id)
```

Comment:
@goern @fridex @pacospace 
As you can I use `get_inspection_status` function from `amun` and it is the place where this step is failing. For me, it feels like if we request the endpoint on swagger api `https://github.com/thoth-station/amun-api/blob/master/amun/api_v1.py#L302` instead, then everything should work OK, should I proceed with that or stay with this one and try to solve?

Reply:
Try to modify the Amun openapi schema so that it accepts None in builds (they can be nullable) and regenarate amun-client.